components/DscMarkdown.tsx

```diff
@@ -0,0 +1,36 @@
+"use client";
+import { unified } from "unified";
+import remarkParse from "remark-parse";
+import remarkGfm from "remark-gfm";
+import remarkRehype from "remark-rehype";
+import rehypeStringify from "rehype-stringify";
+import clsx from "clsx";
+import React, { useEffect, useState } from "react";
+export default function DscMarkdown(props: {
+  children: string;
+  className?: string;
+}) {
+  const [processedContent, setProcessedContent] = useState("");
+
+  useEffect(() => {
+    const processMarkdown = async () => {
+      const result = await unified()
+        .use(remarkParse)
+        .use(remarkGfm)
+        .use(remarkRehype)
+        .use(rehypeStringify)
+        .process(props.children || "");
+
+      setProcessedContent(result.toString());
+    };
+    processMarkdown();
+  }, [props.children]);
+  return (
+    <div className="prose font-inter text-sm leading-relaxed dark:prose-invert sm:text-base xl:text-left">
+      <div
+        className={clsx(props.className ?? "")}
+        dangerouslySetInnerHTML={{ __html: processedContent.toString() }}
+      />
+    </div>
+  );
+}
```

Comment:
The old one is the server side component and 
GithuHub Discussion is client-side comp. So the server side component will not work inside the client component, right?
So I created a new clients side comp , I know double is not needed, that's why if you approve it , I will  make the Markdwon.tsx to the client side .
I mentioned in PR description it is experimental

Reply:
You can render server component inside a client component (as long as the props are serializable)

Ref: https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#interleaving-server-and-client-components

Reply:
Also, as mentioned in other reviews, we'd be moving away from discussions being a client component by not requiring to use a state and useEffect. So this wouldn't be an issue

Reply:
There is still issue because as GithubDiscussion.tsx is still clientcomponent right and we have to pass the discription in the markdwon as children and description will render condtionally for readmore and readless.

Reply:
Like mentioned previously, you can interleave client and server components. Refer the link mentioned above